### PR TITLE
[Remove Stream/Optional usages: headless/headless-admin-content]Lps 178726 3 27

### PR DESCRIPTION
**Pull request by slnn**
https://issues.liferay.com/browse/LPS-178726

### CODE REVIEW

**Comment by Kvale1733** [comment]
ci:test:headless

File: modules/apps/headless/headless-admin-content/headless-admin-content-impl/src/main/java/com/liferay/headless/admin/content/internal/dto/v1_0/util/CreatorUtil.java [hunk 1]
```diff
@@ -62,17 +59,7 @@ public static Creator toCreator(
...
-							).map(
-								parameters -> parameters.getFirst(
-									"nestedFields")
-							).map(
-								fields -> fields.contains("profileURL")
-							).orElse(
-								false
-							)) {
-
+						if (_containProfileURL(uriInfo)) {
```
[the remark this answers is not in the document]
Comment (dantewang) [reply modules/apps/headless/headless-admin-content/headless-admin-content-impl/src/main/java/com/liferay/headless/admin/content/internal/dto/v1_0/util/CreatorUtil.java:62]: There are multiple cases like this.

File: modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/dto/v1_0/converter/StructuredContentDTOConverter.java [hunk 2]
```diff
@@ -232,6 +232,23 @@ private Map<Locale, String> _filterDescriptionMap(
 		);
 	}
 
+	private boolean _notContainRenderedContentValue(UriInfo uriInfo) {
```
Comment (dantewang) [inline modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/dto/v1_0/converter/StructuredContentDTOConverter.java:235]: @slnn Usually we don't just add a method called "notContainXXX". Usually the name would be "containsXXX".

Comment (dantewang) [reply modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/dto/v1_0/converter/StructuredContentDTOConverter.java:235]: @slnn It seems there are a lot of such methods getting the parameters map from uriInfo, gets a value from the map, and then checks whether the value contains something.

Could you see if it's possible to extract a common method somewhere to simplify the code?

File: modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/dto/v1_0/converter/StructuredContentDTOConverter.java [hunk 3]
```diff
@@ -232,6 +232,23 @@ private Map<Locale, String> _filterDescriptionMap(
...
+		MultivaluedMap<String, String> parameters =
+			uriInfo.getQueryParameters();
+
+		if ((parameters == null) || parameters.isEmpty()) {
```
Comment (dantewang) [inline modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/dto/v1_0/converter/StructuredContentDTOConverter.java:239]: @slnn Is it possible for this parameter map to be null?

**Review by dantewang — CHANGES_REQUESTED** [review]

File: modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/dto/v1_0/converter/StructuredContentDTOConverter.java (hunk 3 above)
Comment (slnn) [reply modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/dto/v1_0/converter/StructuredContentDTOConverter.java:239]: Hi @dantewang

The parameter map is possible to be null, please see the implementation for getQueryParameters() in HeadlessDiscoveryOpenAPIResourceImpl which is an implementation of UriInfo

```
@Override
			public MultivaluedMap<String, String> getQueryParameters() {
				return null;
			}
```

**Pull request closed without merging**